Микросетки на флексах для всех страниц

Lyezik opened the pull request:
---
:mortar_board: [Микросетки на флексах для всех страниц](https://up.htmlacademy.ru/htmlcss/34/user/2097541/tasks/8)

:boom: https://htmlacademy-htmlcss.github.io/2097541-sedona-34/6/

Lyezik committed:
Корректирует сетку шапки

Lyezik committed:
Делает ссылку логотипа блочной+ форматирование

Lyezik committed:
Переносит нижний отступ с навигации на хедер

Lyezik committed:
Корректирует сетку секции с поиском отеля и вводную часть преимуществ

Lyezik committed:
Корректирует сетку футера

Lyezik committed:
Корректирует сетку первого преимущества

Lyezik committed:
Корректирует сетку преимуществ

Lyezik committed:
Переносит изменения разметки с главной страницы на страницу каталога

Lyezik committed:
Корректирует шапку каталога

alex-koshara commented on catalog.html:
```diff
@@ -8,12 +8,12 @@
         <img class="logo-img" src="images/logo.svg" width="139" height="70" alt="Логотип Седоны">
       </a>
-      <ul class="navigation-list">
+      <ul class="navigation-list navigation-pages-wrapper">
```
А для чего эти обёртки нужны?

Lyezik replied:
Я её ввел, чтобы указать именно на ссылки перехода на другие страницы. 
```
.navigation-pages-wrapper .navigation-item {
  padding: 20px 17px;
}
```
И не стал для этих navigation-item вводить новый класс, т.к. ".navigation-item" и так нигде больше не задействован. Да и в случае чего можно будет, например, закрепить ширину этой обертки, чтобы в строку помещалось по 3-4 ссылки(если вдруг их много будет)

alex-koshara replied:
Давай этот момент на консультации обсудим. Заодно под запись расскажу что да как в целом?

alex-koshara commented on catalog.html:
```diff
@@ -8,12 +8,12 @@
 
 <body>
-  <header class="page-header">
+  <header class="catalog-header">
```
Хедер - это повторяющаяся часть на других страницах, почему решил переименовать?

Lyezik replied:
Вообще сделал так, чтобы настроить расстояние между хедером и мейном, т.к. на странице каталога он отличается. Но вот про то, что хедер повторяющийся я не подумал. Крч поменял, чтобы этот отступ достигался за счет мейна, а не хедера

Lyezik replied:
Крч, у меня был не актуальный макет) И там расстояние между шапкой и главной частью разное было на разных страницах. Но вот про то, что хедер повторяющийся, я не подумал. Сейчас поменял, чтобы этот отступ достигался за счет мейна, а не хедера. Ну и обновил версию макета, теперь буду следить за этим)

alex-koshara commented on styles/styles.css:
```diff
@@ -209,47 +308,99 @@ body {
+.catalog-header {
+  font-weight: 700;
+  color: black;
+  margin: 0 auto -5px;
```
Старайся по минимуму использовать отрицательные маржины

Lyezik committed:
Меняет отступ между хедером и мейном за счет мейна

Lyezik committed:
Корректирует расстояние между header и main

Lyezik committed:
Корректирует сетку каталога

За исключением фильтра, карточек и секции рассылкой

Lyezik committed:
Поправляет оформление секции с рассылкой в каталоге

Lyezik committed:
Корректирует форму для отправки почты

Lyezik committed:
Добавляет отдельный класс для центровки хедера, мейна, футера и в навигации по бокам делает паддинги

Lyezik committed:
Переименовывает класс, указывающий на текущую страницу

Для большей понятности

Lyezik committed:
Добавляет отдельный класс для центровки хедера, мейна, футера

Lyezik committed:
Делает, чтобы списки в навигации не слиплись в случае переполнения

Lyezik committed:
Убирает минимальную высоту в навигации

Lyezik committed:
Реализует пересечение хедера и основной части за счет хедера

Lyezik committed:
Создает внутренние отступы в заголовке преимеществ и делает, чтобы иконки находились в паддинге своего родителя

Lyezik committed:
Добавляе тотступы по бокам в секции поиска гостиницы

Lyezik committed:
Добавляет общие паддинги в футер и убирает лишние марджины

Lyezik committed:
Корректирует страницу каталога

Lyezik committed:
Форматирование

Lyezik committed:
Исправляет ошибки

1. марджин после фильтра в 5 пх заменить на паддинг
2. ссылки которые для вида показа карточек обернуть в див
3. боковые паддинги пагинации уменьшить
4. обертка пагинации лишняя

alex-koshara commented on styles/styles.css:
```diff
@@ -44,27 +44,31 @@ body {
   overflow: hidden;
 }
 
+.centering-container {
```
Можно просто `container`

alex-koshara commented on styles/styles.css:
```diff
@@ -203,12 +214,11 @@ body {
 
 .advantages-city-item {
-  padding: 80.5px 85px 79.5px;
+  padding: 182.5px 85px 79.5px;
```
Не нужно доли пикселей писать, округли)
